RELEASE.md

```diff
@@ -0,0 +1,66 @@
+# Release Process
+
+The process of creating a release is described in this document.
+Note: below we list the manual steps to create a new release, later some of them will be automated by GitHub actions.
+
+Replace `X.Y.Z` with the version to be released.
+
+## 1. Update the .make.versions file
+
+**Question:** should we use the same versions for release, and for shared python modules and images, such as 
+`DPK_LIB_VERSION`, `DPK_LIB_KFP_VERSION`, `DPK_LIB_KFP_VERSION_v2`, `DPK_LIB_KFP_SHARED`, `KFP_DOCKER_VERSION`, `KFP_DOCKER_VERSION_v2`?
+If YES, maybe we can combine all these variables into a single one.
```

Comment:
Does it makes sense to have `0.0.0` tag to all in dev branch and update it to the release tag upon creating a new release in the release branch? there is no meaning to the tag version in dev as we build everything locally.

Reply:
BTW I think that the process of updating the versions in .make.versions should be done in the release branch so the .make.versions in dev branch should remain `0.0.0`...

Reply:
> there is no meaning to the tag version in dev as we build everything locally.

it is not exactly true, for pipeline tests on a real cluster we need relevant images.